transaction/src/manifest/e2e.rs

```diff
@@ -240,16 +248,34 @@ CALL_METHOD
     "aliases"
     Enum(0u8)
     Enum(0u8)
-    Enum(1u8, "hello")
-    Enum(1u8, "hello")
-    Enum(0u8, "test")
-    Enum(0u8, "test")
-    Enum(1u8, "test123")
-    Enum(1u8, "test123")
+    Enum(1u8,
```

Comment:
I think that it would be nice if the indented formatting for enums with fields was:
```rust
Enum(
    0u8,
    "hello",
    "world",
)
```

what do you think?

Reply:
I think the enum discriminator is conceptually different from the associated values, and more coupled with enum variant, so was put besides `Enum(`. And you've been using this syntax for many of the examples, which I quite like.

But I'm happy to change it if you think the other way is more appropriate. 

_(Another crazy idea is to make it `Enum<5u8>()`)_

Reply:
Agreed on the variant id being different from the fields of the enum. Maybe one way to sort of distinguish these two things is by:

```
Enum(
    0u8, // Variant id
    [    // fields
        "hello",
        "world",
    ]
)
```

Also, I like crazy ideas and would never say no to them :D

Reply:
I like the proposed from a structure wise, but I feel it's not very concise (depth increases by 2, but in fact is 1).

```
Enum(
    0u8,
    []
)

Enum(
    0u8,
    [
         A,
         B,
         C
    ]
)
```

Will give the crazy idea a try.

```
Enum<0u8>()

Enum<1u8>(
    A,
    B,
    C
)
```

Reply:
Yup happy with the "generics" syntax for enum variant ids and also happy with the way that enums are currently formatted :)

Reply:
Updated. Please re-review when you're free. :)